### PR TITLE
[W-11132253 / W-11091132] - Cancel recurring donation

### PR DESCRIPTION
**Pull request by maurimart**
[[Technical Design Review] W-11132253 - NPSP](https://salesforce.quip.com/PLdcAM3askNg)
[[Technical Design Review] W-11091132 - NPSP](https://salesforce.quip.com/MhvnAiCjaNS6)

# Critical Changes

Add cancel recurring donation handler for modal and method for RD2_ETableController.
Disable recurring donation actions based on rd status.

# Changes

### CODE REVIEW

**Comment by maurimart** [comment]
[W-11132253](https://gus.lightning.force.com/a07EE00000wrbMXYAY)

**Review by gerardorodriguezsf — CHANGES_REQUESTED** [review]
- Update apex method to use the recurring donation record.
- Update method to just upsert the record.
- Check CRUD permissions.
- Update Cancel LWC to get the Recurring Donation record, update the Status, and send to apex method to be upserted.
- Prevent duplicated actions due to the rendered actions.

**Comment by maurimart** [comment]
> * Prevent

Changes applied!!!

**Review by gerardorodriguezsf — CHANGES_REQUESTED** [review]
Missing exception for missing permissions:
`} else {
                throw new UTIL_Permissions.InsufficientPermissionException(Label.commonInsufficientPermissions);
            }`

**Review by gerardorodriguezsf — APPROVED** [review]

**Comment by maurimart** [comment]
[W-11091132](https://gus.lightning.force.com/a07EE00000wFnf5YAC)

File: force-app/main/default/classes/RD2_ETableController.cls
Comment (lparrott) [inline force-app/main/default/classes/RD2_ETableController.cls:110]: Description and return are wrong here.

File: force-app/main/default/labels/CustomLabels.labels-meta.xml
Comment (lparrott) [inline force-app/main/default/labels/CustomLabels.labels-meta.xml:1866]: Are these Labels all approved? "Contact us if you need help." doesn't seem like something we'd normally put in a message 🤷

File: force-app/main/default/lwc/rd2RecurringDonation/rd2RecurringDonation.js
Comment (lparrott) [inline force-app/main/default/lwc/rd2RecurringDonation/rd2RecurringDonation.js:309]: Remove console.log

**Review by lparrott — CHANGES_REQUESTED** [review]

File: force-app/main/default/lwc/rd2RecurringDonation/rd2RecurringDonation.js
Comment (maurimart) [reply force-app/main/default/lwc/rd2RecurringDonation/rd2RecurringDonation.js]: removed, thanks!

File: force-app/main/default/classes/RD2_ETableController.cls
Comment (maurimart) [reply force-app/main/default/classes/RD2_ETableController.cls:110]: Changed!

File: force-app/main/default/labels/CustomLabels.labels-meta.xml
Comment (maurimart) [reply force-app/main/default/labels/CustomLabels.labels-meta.xml:1866]: @lparrott the labels are defined in this document https://salesforce.quip.com/kRYaAjEE8aHr the text is as described.

File: force-app/main/default/lwc/rd2RecurringDonation/rd2RecurringDonation.html
Comment (force2b) [inline force-app/main/default/lwc/rd2RecurringDonation/rd2RecurringDonation.html:148]: hardcoded text (though maybe this was fixed in another PR?)

Comment (maurimart) [reply force-app/main/default/lwc/rd2RecurringDonation/rd2RecurringDonation.html:148]: Yes, this was fixed in another PR, would we still need to fix it here?

Comment (force2b) [reply force-app/main/default/lwc/rd2RecurringDonation/rd2RecurringDonation.html:148]: If it's a small change then I recommend it, especially if this is going to be UI tested before merging. It also ensures that future merges don't accidentally overwrite the fix to the label with hardcoded text.

Comment (maurimart) [reply force-app/main/default/lwc/rd2RecurringDonation/rd2RecurringDonation.html]: Fixed!!

File: force-app/main/default/lwc/stopRecurringDonationModal/stopRecurringDonationModal.html
Comment (force2b) [inline force-app/main/default/lwc/stopRecurringDonationModal/stopRecurringDonationModal.html:18]: hardcoded text here

**Review by force2b — CHANGES_REQUESTED** [review]

File: force-app/main/default/lwc/stopRecurringDonationModal/stopRecurringDonationModal.html
Comment (maurimart) [reply force-app/main/default/lwc/stopRecurringDonationModal/stopRecurringDonationModal.html:18]: Changed!

**Review by lparrott — APPROVED** [review]

File: force-app/main/default/labels/CustomLabels.labels-meta.xml
Comment (force2b) [inline force-app/main/default/labels/CustomLabels.labels-meta.xml:6161]: Should this be "Cancel and Close"?
Has this label been reviewed by docs?

Comment (maurimart) [reply force-app/main/default/labels/CustomLabels.labels-meta.xml:6161]: yes, this corresponds to the modal's close icon, as defined in the slds documentation

**Pull request merged**
